### DISCUSSION TITLE
Noob Question: Do all the charging control methods work by manipulating temperature sensing?

### DISCUSSION THREAD

**Discussion by dbza** [VR-25/acc#101] [Q&A] [answered]
The google photos app has a "device too hot" condition that prevents automatic backups. (https://support.google.com/photos/thread/97451457/cant-backup-my-photos-because-device-is-too-hot?hl=en)

 I use acc on an old pixel device (acc -sc 250)  I am wondering if the high temperature reported is reported by  acc? If so is there any way to keep using acc to save my weak battery and still be able to run backups.

**Comment by VR-25** [comment, marked as answer]
Hey there,
Temperature is reported by the kernel (charging drivers).
Were you able to figure out the temperature threshold at which "device too hot" kicks in?
You may try freezing the battery service by setting a custom battery level - before that threshold.
This will keep android from getting battery status updates (including temperature) from the kernel.
```
$ su
# # fake 50%
# dumpsys battery set level 50
# # undo
# dumpsys battery reset
```

**Reply by dbza** [reply]
Thanks for taking the time to reply!
 It turned out the warning happens even without acc running. Threshold is around 35-40 degrees C, I found a workaround (more external cooling) so all good now!
